go/internal/a2a/a2a_registrar.go

```diff
@@ -0,0 +1,171 @@
+package a2a
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"reflect"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/kagent-dev/kagent/go/api/v1alpha2"
+	agent_translator "github.com/kagent-dev/kagent/go/internal/controller/translator/agent"
+	authimpl "github.com/kagent-dev/kagent/go/internal/httpserver/auth"
+	common "github.com/kagent-dev/kagent/go/internal/utils"
+	"github.com/kagent-dev/kagent/go/pkg/auth"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
+	crcache "sigs.k8s.io/controller-runtime/pkg/cache"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	a2aclient "trpc.group/trpc-go/trpc-a2a-go/client"
+)
+
+type A2ARegistrar struct {
+	cache          crcache.Cache
+	translator     agent_translator.AdkApiTranslator
+	handlerMux     A2AHandlerMux
+	a2aBaseUrl     string
+	authenticator  auth.AuthProvider
+	a2aBaseOptions []a2aclient.Option
+}
+
+var _ manager.Runnable = (*A2ARegistrar)(nil)
+
+func NewA2ARegistrar(
+	cache crcache.Cache,
+	translator agent_translator.AdkApiTranslator,
+	mux A2AHandlerMux,
+	a2aBaseUrl string,
+	authenticator auth.AuthProvider,
+	streamingMaxBuf int,
+	streamingInitialBuf int,
+	streamingTimeout time.Duration,
+) *A2ARegistrar {
+	reg := &A2ARegistrar{
+		cache:         cache,
+		translator:    translator,
+		handlerMux:    mux,
+		a2aBaseUrl:    a2aBaseUrl,
+		authenticator: authenticator,
+		a2aBaseOptions: []a2aclient.Option{
+			a2aclient.WithTimeout(streamingTimeout),
+			a2aclient.WithBuffer(streamingInitialBuf, streamingMaxBuf),
+			debugOpt(),
+		},
+	}
+
+	return reg
+}
+
+func (a *A2ARegistrar) NeedLeaderElection() bool {
+	return false
+}
+
+func (a *A2ARegistrar) Start(ctx context.Context) error {
+	log := ctrllog.FromContext(ctx).WithName("a2a-registrar")
+
+	informer, err := a.cache.GetInformer(ctx, &v1alpha2.Agent{})
+	if err != nil {
+		return fmt.Errorf("failed to get cache informer: %w", err)
+	}
+
+	if _, err := informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
```

Comment:
Is there a reason you used the cache directly here instead of creating a `Controller` like the rest of the k8s watchers? I usually prefer consistency across the various watchers so the codebase is easier to grok, but definitely open to this if there's a good reason.